packages/blaze-tools/tojs.js

```diff
@@ -89,19 +89,19 @@ ToJSVisitor.def({
       tagSymbol = 'HTML.getTag(' + toJSLiteral(name) + ')';
     }
 
-    var attrsArray = null;
+    let attrsArray = null;
     if (attrs) {
       attrsArray = [];
-      var needsHTMLAttrs = false;
+      let needsHTMLAttrs = false;
```

Comment:
needsHTMLAttrs is not defined in the tests. I assume this is a scoping issue and previously `needsHTMLAttrs` was simply wrong scoped and used outside it's defined scope and since it was a `var` this was possible.

I think this whole block of loops need some more refactoring effort. Maybe we can split it into several methods to better handle scoping?

Reply:
Agreed, I was going just for the minimum reproduction to move things forward one step at a time.